feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/radio.c

```diff
@@ -300,24 +307,26 @@ bool target_radio_config_set2(const struct schema_Wifi_Radio_Config *rconf,
 		int tx_ant_avail;
 		tx_ant_avail=phy_get_tx_available_antenna(phy);
 
-		if ((rconf->tx_chainmask & tx_ant_avail) != rconf->tx_chainmask) {
+		if (rconf->tx_chainmask == 0) {
 			blobmsg_add_u32(&b, "txantenna", tx_ant_avail);
-			LOGN("maximum number of tx antennae available to use:%d",tx_ant_avail);
-		}
-		else
+		} else if ((rconf->tx_chainmask & tx_ant_avail) != rconf->tx_chainmask) {
+			blobmsg_add_u32(&b, "txantenna", tx_ant_avail);
```

Comment:
Can we club the two conditions (if and else if) with an || since both are performing same action?

Reply:
Leaving separate because first is 'Auto' case.  Second is defence against misconfiguration.